### PR TITLE
feat(onyx-434): Add 'details' text input to create/update alert UI

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-434]

### Description

Add `details` text input to create/update alert UI

| Create form | Edit form |
|---|---|
| Simulator Screenshot - iPhone 15 Pro - 2023-10-24 at 11 52 19 | Simulator Screenshot - iPhone 15 Pro - 2023-10-24 at 11 53 40 |

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-  Add 'details' text input to create/update alert UI - @nickskalkin 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-434]: https://artsyproduct.atlassian.net/browse/ONYX-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/store/config/features.ts [hunk 1]
```diff
@@ -229,6 +229,11 @@ export const features: { [key: string]: FeatureDescriptor } = {
     readyForRelease: false,
     showInDevMenu: true,
   },
+  AREnableAlertDetailsInput: {
```
Comment (nickskalkin) [inline src/app/store/config/features.ts:232]: Not sure if we need a feature flag for such a small change. Let me know what you think - I'll either delete it or make it ready for release

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Add 'details' text input to create/update alert UI - @nickskalkin)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 750352138f22e1a2fbcd8bc32d38291e157da0de
</p>

**Review by dariakoko — DISMISSED** [review]

**Review by MounirDhahri — DISMISSED** [review]

**Review by dblandin — DISMISSED** [review]
LGTM

File: src/app/store/config/features.ts (hunk 1 above)
Comment (nickskalkin) [reply src/app/store/config/features.ts:232]: agree! PRed echo update: https://github.com/artsy/echo/pull/557

**Pull request merged**